Makefile

```diff
@@ -586,8 +586,12 @@ test-hash-seed: all
 
 .PHONY: test-doc
 test-doc: doc-only ## Builds, lints, and verifies the docs.
-	$(MAKE) lint
-	$(PYTHON) tools/test.py $(PARALLEL_ARGS) $(CI_DOC)
+	@if [ "$(shell $(node_use_openssl))" != "true" ]; then \
```

Comment:
Where does this variable come from? Am I supposed to invoke this target with `make test-doc node_use_openssl=true`?

Can I suggest this alternative patch? That makes configure.py write it to config.mk.
```diff
diff --git a/Makefile b/Makefile
index 860c27676c..74145008e5 100644
--- a/Makefile
+++ b/Makefile
@@ -586,8 +586,12 @@ test-hash-seed: all
 
 .PHONY: test-doc
 test-doc: doc-only ## Builds, lints, and verifies the docs.
+ifdef NODE_WITHOUT_SSL
+       @echo "Skipping test-doc (no crypto)"
+else
        $(MAKE) lint
        $(PYTHON) tools/test.py $(PARALLEL_ARGS) $(CI_DOC)
+endif
 
 test-known-issues: all
        $(PYTHON) tools/test.py $(PARALLEL_ARGS) known_issues
diff --git a/configure.py b/configure.py
index 16ee259dc2..5e1140138f 100755
--- a/configure.py
+++ b/configure.py
@@ -1651,6 +1651,9 @@ if options.prefix:
 if options.use_ninja:
   config['BUILD_WITH'] = 'ninja'
 
+if options.without_ssl:
+  config['NODE_WITHOUT_SSL'] = '1'
+
 config_lines = ['='.join((k,v)) for k,v in config.items()]
 # Add a blank string to get a blank line at the end.
 config_lines += ['']
```

Reply:
> Where does this variable come from?

It is defined in the [Makefile](https://github.com/nodejs/node/blob/43a51708589ac789ce08beaeb49d6d778dfbdc49/Makefile#L364). 

>Am I supposed to invoke this target with make test-doc node_use_openssl=true?

I've only used `make test-doc` without specifying anything (apart from ./configure --without-ssl that is).

>Can I suggest this alternative patch? That makes configure.py write it to config.mk.

Always! I've got not problem with that patch but I should probably let the build group have their say on this. I think I recall doing something similar before but that did not cover all usages (CI, test environments etc).

Reply:
Ah, never mind. I didn't realize it's not necessarily using the node from out/Release. Please ignore my previous comment. :-)